### PR TITLE
feat(keel): add support for managed delivery

### PR DESCRIPTION
**Pull request by nimakaviani**

### CODE REVIEW

File: keel/base/deployment.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  namespace: spinnaker
+spec:
+  replicas: 1
+  template:
+    spec:
+      containers:
+      - env:
+        - name: JAVA_OPTS
+          value: -XX:MaxRAMPercentage=50.0
+        image: docker.io/nimak/spinnaker-keel:latest
```
Comment (nimakaviani) [inline keel/base/deployment.yml]: the image needs to be pushed to the spinnaker owned repo, so this will have to change.

**Comment by nimakaviani** [comment]
As mentioned in the comment, the image needs to be pushed to the spinnaker registry. the PR is to receive early feedback and also to get help with how and where to push the image.

**Review by maggieneterval — COMMENTED** [review]
Thanks @nimakaviani, this looks 💯  other than needing to push the `keel` image to `spinnaker-marketplace`. What is the current release process for `keel`? Is the plan to integrate it into the top-level Spinnaker build and release process, or just have some GitHub action that publishes the image to `spinnaker-marketplace`? If this conversation isn't happening elsewhere already, might be worth posting in the #build-process Slack channel. But otherwise, this looks good, so excited that `keel` will be deployable using `kleat`!

File: README.md [hunk 2]
```diff
@@ -187,6 +186,28 @@ services:
...
+
+Since the default
+[service discovery file](/core/service-discovery/spinnaker.yml) defaults
+Keel to disabled, you will need to configure a service discovery overrides file
+in your fork of spinnaker-config with the following block:
+
+```
+services:
+  keel:
+    enabled: true
```
Comment (maggieneterval) [inline README.md]: Did we determine that we don't actually need this and `kleat` will write out the `enabled` block when managed delivery is enabled in users' halconfigs?

Comment (nimakaviani) [reply README.md]: yeah I believe so. I am waiting on answer to your question from @ezimanyi on the other thread before nuking it for Keel and Kayenta.

**Comment by nimakaviani** [comment]
> What is the current release process for keel? Is the plan to integrate it into the top-level Spinnaker build and release process, or just have some GitHub action that publishes the image to spinnaker-marketplace? If this conversation isn't happening elsewhere already, might be worth posting in the #build-process Slack channel. 

great point. I will raise the question in the build process channel.

**Comment by nimakaviani** [comment]
got the docs fixed following [this comment](https://github.com/spinnaker/kleat/pull/145#discussion_r502440859).

File: README.md (hunk 2 above)
Comment (maggieneterval) [reply README.md]: Thanks for updating this!

File: keel/base/deployment.yml (hunk 1 above)
Comment (maggieneterval) [reply keel/base/deployment.yml]: While I don't want to unnecessarily block `kleat` users who want to try deploying `keel`, I don't think I feel comfortable merging this as-is if users deploying this config will be pulling the latest `keel` from a personal, non-Spinnaker image repository. Is `keel` close to being published in the Spinnaker-owned repo? If we do want to move forward with merging this as-is, can we better communicate to users that this image is not yet maintained in the Spinnaker-owned repo and it's a use-at-your-own-risk situation?

Comment (nimakaviani) [reply keel/base/deployment.yml]: yeah @maggieneterval I agree and it makes total sense. I am working with the buildcops [here](https://spinnakerteam.slack.com/archives/CKXFWFB53/p1602529528020000) to make this happen. Let's have this PR blocked on building and pushing the image to the official spinnaker registry.

**Review by german-muzquiz — APPROVED** [review]
Approving this since it was already reviewed and only missing the docker image update

**Pull request merged**